### DISCUSSION TITLE
Visualization of the Generated simulation

### DISCUSSION THREAD

**Discussion by jais001** [ViennaTools/ViennaPS#89] [Q&A] [answered]
Hi,

I have started using the ViennaPS for the simulation of hole etching and Trench deposition. I am using Python, and followed the examples added in:

1. I have tried a sample for Trench deposition. It is throwing an error :

```
TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. viennaps2d.viennaps2d.Process()
    2. viennaps2d.viennaps2d.Process(domain: viennaps::Domain<double, 2>)
    3. viennaps2d.viennaps2d.Process(domain: viennaps::Domain<double, 2>, model: viennaps2d.viennaps2d.ProcessModel, duration: float
```

So I used trench deposition geometric, where the script generates a final.vtp file. I have used VTK library for the visualization process and converted it to an image. The Image I obtained is actually black and white which is totally different from the image in the ReadMe. I have attached a sample image. Can I know if there is a different method for the simulation visualization or are there any configuration changes that I need to make to obtain the images as seen in readme?

code: https://github.com/ViennaTools/ViennaPS/tree/master/examples/trenchDepositionGeometric

generated image:
![trench_deposition_final](https://github.com/user-attachments/assets/d8657559-cf40-4e5d-92a1-8e9627f3a227)

image in readme:
![deposition](https://github.com/user-attachments/assets/3c995c6b-5b42-4969-a39c-72a31adeb114)

3. The same happens for hole etching:
     code: https://github.com/ViennaTools/ViennaPS/tree/master/examples/holeEtching

The 3D image being generated is:
![final](https://github.com/user-attachments/assets/ba30eb89-d276-4ae3-8f79-c1571311093c)
 This is also totally different from the 3D diagram in readme. Also While executing the simulation script I couldn't find any config for saving the byproduct images whose name starts from: `SF6O2Etching_0.vtp` which is causing the byproduct images being spammed in the main folder.

Any assistance with these issues would be greatly appreciated. Thank you!

**Comment by tobre1** [comment, marked as answer]
Hi there!

For visualization purposes, I recommend using ParaView. It's specifically designed to handle geometries generated from the VTK library, making it an excellent tool for viewing and analyzing your data. You can download ParaView for most operating systems directly from their [official website](https://www.paraview.org/download/).

The images shown in the README were created using the [trenchDeposition](https://github.com/ViennaTools/ViennaPS/tree/master/examples/trenchDeposition) example with varying parameters for the sticking coefficient. I recently ran this example using the ViennaPS Python package (version 3.1.0) installed via pip, and it worked without any issues. If you encounter any errors while running this or other examples, please provide details about the version you're using and the specific error message you received. This will help in diagnosing the problem. I also recommend using the latest version of the ViennaPS package from [PyPI](https://pypi.org/project/ViennaPS/) to ensure compatibility and access to the latest features.

If you'd like to save intermediate results during the simulation, you can adjust the Logger settings in the code. By setting `vps.LogLevel.INTERMEDIATE`, you'll be able to generate and save point clouds at each time step, giving you a detailed view of the process as it progresses.

**Comment by jais001** [comment]
@tobre1 Hi, I have checked Paraview, but this is a GUI-based application, right? Currently, we are planning on using ViennaPS to generate simulations in our rest API. So I would like to know if there is a way to provide it in our client-server architecture. Are there any Python packages for converting this VTP output as a 2D or 3D image? Or are there any react packages for rendering vtp files, so that I can supply the generated VTP file to server side?

Any assistance would be greatly appreciated. Thank you!

**Comment by jais001** [comment]
@tobre1 Thanks for the info. I have found Pyvista as a suitable package for conversion of the simulation to 3D images using Python.
